pkg/auth/ldaputil/testclient/testclientconfig.go

```diff
@@ -0,0 +1,37 @@
+package testclient
+
+import "github.com/openshift/origin/pkg/auth/ldaputil/ldapclient"
+
+// fakeConfig regurgitates internal state in order to conform to Config
+type fakeConfig struct {
+	client       ldapclient.Client
+	bindDN       string
+	bindPassword string
+	ldapURL      string
+}
+
+// NewConfig creates a new Config impl that regurgitates the given data
+func NewConfig(client ldapclient.Client, bindDN, bindPassword, ldapURL string) ldapclient.Config {
+	return &fakeConfig{
+		client:       client,
+		bindDN:       bindDN,
+		bindPassword: bindPassword,
+		ldapURL:      ldapURL,
+	}
+}
+
+func NewConnectingConfig(client ldapclient.Client) ldapclient.Config {
```

Comment:
Pass the empty strings.  It makes it easier to follow.

On Fri, Oct 30, 2015 at 8:41 AM, Steve Kuznetsov <notifications@github.com>
wrote:

> In pkg/auth/ldaputil/testclient/testclientconfig.go
> <https://github.com/openshift/origin/pull/5145#discussion_r43497579>:
>
> > +	bindDN       string
> > +	bindPassword string
> > +	ldapURL      string
> > +}
> > +
> > +// NewConfig creates a new Config impl that regurgitates the given data
> > +func NewConfig(client ldapclient.Client, bindDN, bindPassword, ldapURL string) ldapclient.Config {
> > +	return &fakeConfig{
> > +		client:       client,
> > +		bindDN:       bindDN,
> > +		bindPassword: bindPassword,
> > +		ldapURL:      ldapURL,
> > +	}
> > +}
> > +
> > +func NewConnectingConfig(client ldapclient.Client) ldapclient.Config {
>
> I didn't like having empty strings passed in all the time to NewConfig,
> is there a better way to do that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5145/files#r43497579>.
>